Source/WebCore/Modules/gamepad/NavigatorGamepad.cpp

```diff
@@ -89,8 +92,46 @@ ExceptionOr<const Vector<RefPtr<Gamepad>>&> NavigatorGamepad::getGamepads(Naviga
     return NavigatorGamepad::from(navigator)->gamepads();
 }
 
+// The UIProcess tracks when a WebPage has recently used gamepads to configure certain behaviors on the page.
+//
+// Once a WebPage notifies the UIProcess that gamepads have been accessed, the UIProcess starts a timer with
+// a short delay, after which it assumes that WebPage is no longer actively using gamepads.
+//
+// This works because of the polling nature of the gamepad API - If a web page is using gamepads it will be
+// accessing them multiple times per second.
+//
+// WebPages need to continuously tell the UIProcess that gamepads have been used recently, but can do so lazily;
+// As long as a WebPage continuously using gamepads notifies the UIProcess at least once during the UIProcess's
+// timer delay, things will work as expected.
+//
+// So it follows: Web processes have this value initialized to be compatible with the UIProcess timer threshold.
+
+static Seconds gamepadsRecentlyAccessedThreshold;
+
+void NavigatorGamepad::setGamepadsRecentlyAccessedThreshold(Seconds threshold)
+{
+    // This value should only initialized once.
+    RELEASE_ASSERT(!gamepadsRecentlyAccessedThreshold);
+    gamepadsRecentlyAccessedThreshold = threshold;
+}
+
+void NavigatorGamepad::maybeNotifyRecentAccess()
+{
+    if (MonotonicTime::now() - m_lastAccessNotificationTime < gamepadsRecentlyAccessedThreshold)
+        return;
```

Comment:
The threshold should probably be moved to WebChromeClient since there could be multiple navigators for the same page, and we don't want duplicate messages from one process as discussed.